src/globals.ts

```diff
@@ -94,7 +94,7 @@ export const removeBitsBuffer = (): void => {
 };
 
 export const timestampToIndex = (timestamp: number): number =>
-    (timestamp / microSecondsPerSecond) * options.samplesPerSecond;
+    Math.floor((timestamp / microSecondsPerSecond) * options.samplesPerSecond);
```

Comment:
The reason for `Math.floor` was that between index 0 and  0.9999999.... there are an infinite amount of timestamps that should give us index 0 as we only have a point at index 0. 

e.g. if we record data every 10ms, thats after 100ms we have 10 data points.
if I would like to get the data at 16ms I would like to return the data at 10ms the data at 20ms might have not been written. 

Interpolation is done in the accumulator and we group point there and calculate the min and max there

not sure if this answers your question

Reply:
At first I thought that this would be completely wrong because the $timestamp$ is in microseconds ($\mu s$), whereas the other values are in $seconds$ notation. However, as stated in my previous comment you the _seconds_ part can be removed by using the last formula $C = S \times 10^6$, which is then just a constant.

We can easily show that it checks out in a simple example:
If we assume samplesPerSecond is set to $100_000$, then there will be a new sample every $10 \mu s$, because $1s = 1,000,000 \mu s$, and divided by the rate will equal $10 \mu s$.

With this in mind imagine the array filling in from the zeroth index:
| index            |  0 |  1   | 2  | 3  |4   | 5  | 6  | ... |
|---|---|---|---|---|---|---|---|---|
|  timestamp |  0 |   10|  20 | 30  |40|  50 | 60  | ... |

Meaning that if we use the formula described by the code at this point, then:

A timestamp of exactly $50 \mu s$ then:

$$ index = { 50  \over 1e6 } \times 100 000  = 50 \times 0,1 = 5 $$

Which also shows that my previous comment was rubbish, as it wouldn't make sense to _floor_ the timestamp 🙃